post007/src/ofApp.cpp

```diff
@@ -1,127 +1,123 @@
 #include "ofApp.h"
+#include "shapes/Triangle.h"
+#include "soo_export.h"
+
+void ofApp::GenerateBalls() {
+  // Setup the common brush parameters for all the balls
+  soo::GenerateTriangleParams brush_triangle_params;
+  soo::GenerateColorParams brush_color_params;
+
+  brush_triangle_params.num_triangles(5)
+      .min_offset_to_center(0)
+      .max_offset_to_center(15);
+  brush_color_params.color(ofColor{0}).alpha(50);
+
+  // Reset the vetor of balls
+  balls_.clear();
+  balls_.reserve(kNumBalls);
+
+  std::generate_n(
+      std::back_inserter(balls_), kNumBalls,
+      [this, &brush_triangle_params, &brush_color_params]() {
+        HairyShape ball;
+
+        // Define the ball contour for the hairy shape, such that:
+        // - the ball is placed inside the bounding box of the seven, but
+        // outside the seven itself (with a specific maximal closeness to it)
+        // - the contour of a ball is defined as a circle with a random radius
+        ofRectangle boundary = seven_.contour_.getBoundingBox();
+        boundary.translate(-boundary.width / 4.f, -boundary.height / 4.f);
+        boundary.scale(1.5);
```

Comment:
this doesn't need to be inside the generator, right? can be done before, outside